src/test/java/com/yelp/nrtsearch/server/luceneserver/facet/RuntimeScriptFacetsTest.java

```diff
@@ -0,0 +1,374 @@
+          switch (source) {
+            case "int":
+              return new IntScript(params, docLookup, context);
+            case "string":
+              return new StringScript(params, docLookup, context);
+            case "map":
+              return new MapScript(params, docLookup, context);
+            case "list":
+              return new ListScript(params, docLookup, context);
+            case "docValue":
+              return new DocValueScript(params, docLookup, context);
+          }
+          throw new IllegalStateException("Unsupported script source: " + source);
+        }
+
+        public static class IntScript extends RuntimeScript {
+
+          public IntScript(
+              Map<String, Object> params, DocLookup docLookup, LeafReaderContext leafContext) {
+            super(params, docLookup, leafContext);
+          }
+
+          @Override
+          public Object execute() {
+            return 2;
+          }
+        }
+
+        public static class StringScript extends RuntimeScript {
+
+          public StringScript(
+              Map<String, Object> params, DocLookup docLookup, LeafReaderContext leafContext) {
+            super(params, docLookup, leafContext);
+          }
+
+          @Override
+          public Object execute() {
+            return "2";
+          }
+        }
+
+        public static class MapScript extends RuntimeScript {
+
+          public MapScript(
+              Map<String, Object> params, DocLookup docLookup, LeafReaderContext leafContext) {
+            super(params, docLookup, leafContext);
+          }
+
+          @Override
+          public Object execute() {
+            return Collections.singletonMap("key", 2.0);
+          }
+        }
+
+        public static class ListScript extends RuntimeScript {
+
+          public ListScript(
+              Map<String, Object> params, DocLookup docLookup, LeafReaderContext leafContext) {
+            super(params, docLookup, leafContext);
+          }
+
+          @Override
+          public Object execute() {
+            List nums = new ArrayList();
+            nums.add("1");
+            nums.add("2");
+            return nums;
+          }
+        }
+
+        public static class DocValueScript extends RuntimeScript {
+
+          public DocValueScript(
+              Map<String, Object> params, DocLookup docLookup, LeafReaderContext leafContext) {
+            super(params, docLookup, leafContext);
+          }
+
+          @Override
+          public Object execute() {
+            Map<String, LoadedDocValues<?>> doc = getDoc();
+            return doc.get("atom_1").get(0) + "_" + doc.get("atom_2").get(0);
+          }
+        }
+      }
+    }
+  }
+
+  protected List<String> getIndices() {
+    return Collections.singletonList(DEFAULT_TEST_INDEX);
+  }
+
+  protected FieldDefRequest getIndexDef(String name) throws IOException {
+    return getFieldsFromResourceFile("/facet/facet_script_facets.json");
```

Comment:
This does not look like the right file name. You also should not need any reference to facets.